### PR TITLE
Migrate to GitHub Actions

### LINKED ISSUE #159: Migrate to GitHub Actions
@seisman @wangliang1989 I'd like to help if you think it's necessary.

### PR DESCRIPTION
**Pull request by core-man**
Fixes #159.

### CODE REVIEW

**Comment by core-man** [comment]
[hypcap](https://www.ctan.org/pkg/hypcap) is required, which is name as `oberdiek` in TEX Live.

**Comment by wangliang1989** [comment]
抱歉，我不知道这方面的内容。我没有管过这些

**Comment by core-man** [comment]
> 抱歉，我不知道这方面的内容。我没有管过这些

That's fine~

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Install LaTeX packages
+        run: |
+          tlmgr install capt-of ctex datetime2 enumitem everysel fancyhdr \
+                        fandol float ms needspace oberdiek parskip sourcecodepro tabulary \
+                        titlesec tracklang ulem upquote varwidth wrapfig \
+                        xcolor xecjk zhnumber
+          sudo cp -r ~/.TinyTeX/texmf-dist/fonts/opentype/adobe/sourcecodepro /usr/share/fonts/opentype
+          fc-cache -f -v
+          ls /usr/share/fonts/opentype
```
Comment (core-man) [inline .github/workflows/deploy.yml]:
```suggestion
          sudo cp -r ~/.TinyTeX/texmf-dist/fonts/opentype/adobe/ /usr/share/fonts/opentype
          fc-cache -f -v
          ls /usr/share/fonts/opentype
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Install LaTeX packages
+        run: |
+          tlmgr install capt-of ctex datetime2 enumitem everysel fancyhdr \
+                        fandol float ms needspace oberdiek parskip sourcecodepro tabulary \
+                        titlesec tracklang ulem upquote varwidth wrapfig \
+                        xcolor xecjk zhnumber
+          sudo cp -r ~/.TinyTeX/texmf-dist/fonts/opentype/adobe/ /usr/share/fonts/opentype
+          fc-cache -f -v
+          ls /usr/share/fonts/opentype
```
Comment (core-man) [inline .github/workflows/deploy.yml]:
```suggestion
```

**Comment by core-man** [comment]
We have to cp fonts to the `/usr/share/fonts`, otherwise there will be a bug. I've no idea why we don't need to do in [cv](https://github.com/seisman/cv/blob/master/.github/workflows/build.yaml)

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt
```
Comment (seisman) [inline .github/workflows/deploy.yml]:
```suggestion
        run: pip install -r requirements.txt
```

**Comment by seisman** [comment]
The current workflow only builds the documentation, but doesn't deploy it.

**Comment by core-man** [comment]
> The current workflow only builds the documentation, but doesn't deploy it.

Yes, will do it later.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Install TinyTeX
+        uses: r-lib/actions/setup-tinytex@v1
+
+      - name: Install LaTeX packages
+        run: |
+          tlmgr install capt-of ctex datetime2 enumitem everysel fancyhdr \
+                        fandol float ms needspace oberdiek parskip sourcecodepro tabulary \
+                        titlesec tracklang ulem upquote varwidth wrapfig \
+                        xcolor xecjk zhnumber
+          sudo cp -r ~/.TinyTeX/texmf-dist/fonts/opentype/adobe/ /usr/share/fonts/opentype
```
Comment (core-man) [inline .github/workflows/deploy.yml:46]:
```suggestion
          # Update the fonts
          sudo cp -r ~/.TinyTeX/texmf-dist/fonts/opentype/adobe/ /usr/share/fonts/opentype
```

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+#
+# Build and deploy the documentation
+#
+name: Deploy
+
+on:
+  pull_request: # enable pull_request for testing
```
Comment (core-man) [inline .github/workflows/deploy.yml]:
```suggestion
  #pull_request: # enable pull_request for testing
```

**Review by seisman — APPROVED** [review]

**Pull request merged**